File: python/test/t_SpaceFilling_std.py
#! /usr/bin/env python

from __future__ import print_function
import openturns as ot

# Defining parameters
dimension = 5
size = 100

# Build OT LHS algorithm
lhs = ot.LHSExperiment(
    ot.ComposedDistribution([ot.Uniform()] * dimension), size)

# Generate design without optimization
design = lhs.generate()

# Defining space fillings
spaceFillingC2 = ot.SpaceFillingC2()
spaceFillingMinDist = ot.SpaceFillingMinDist()
spaceFillingPhiP = ot.SpaceFillingPhiP()
spaceFillingPhiP50 = ot.SpaceFillingPhiP(50)

# print the criteria on this design
print("C2=%f MinDist=%f PhiP=%f, PhiP(50)=%f" %
      tuple([sf.evaluate(design) for sf in [spaceFillingC2, spaceFillingMinDist, spaceFillingPhiP, spaceFillingPhiP50]]))

File: lib/test/t_SpaceFilling_std.cxx
#include <iostream>
// OT includes
#include <openturns/OT.hxx>

using namespace OT;

int main(int, char *[])
{
  const UnsignedInteger dimension(5);
  const UnsignedInteger size(100);

  Collection<Distribution> distributions(dimension);
  for(UnsignedInteger i = 0; i < dimension; ++i)
    distributions[i] = Uniform();

  LHSExperiment lhs(JointDistribution(distributions), size);
  Sample design(lhs.generate());

  // Defining space fillings
  SpaceFillingC2 spaceFillingC2;
  SpaceFillingMinDist spaceFillingMinDist;
  SpaceFillingPhiP spaceFillingPhiP;
  SpaceFillingPhiP spaceFillingPhiP50(50);

  // Print the criteria on this design
  std::cout << "C2=" << spaceFillingC2.evaluate(design)
            << ", MinDist=" << spaceFillingMinDist.evaluate(design)
            << ", Phip=" << spaceFillingPhiP.evaluate(design)
            << ", Phip(50)=" << spaceFillingPhiP50.evaluate(design)
            << std::endl;
  return 0;
}
